rust/src/plutus.rs

```diff
@@ -475,6 +475,13 @@ impl PlutusList {
         }
     }
 
+    pub(crate) fn clone_definite(&self) -> Self {
+        Self {
+            elems: self.elems.clone(),
+            definite_encoding: Some(true),
+        }
+    }
```

Comment:
@rooooooooob , I am not 100% sure why indefinite serialization been implemented for the plutus list in the first place, and whether it is correct to be used in any other case apart from the `hash_script_data` function. It might be that it SHOULD be indefinite everywhere else, but because of an error in the hashing - it has to be definite in `hash_script_data`. Or it could be that it actually should be definite everywhere at all, and then maybe the `Serialize` implementation for the `PlutusList` must be changed to always do definite serializing. 

So far it seems that plutus data hashes are correct between the serlib and CLI, we'll be rechecking it for `PlutusList` now

Reply:
Seems to be resolved by this: https://github.com/Emurgo/cardano-serialization-lib/pull/373#issuecomment-1039513926